tool/python/examples/train_cifar10.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python
+
+#/************************************************************
+#*
+#* Licensed to the Apache Software Foundation (ASF) under one
+#* or more contributor license agreements.  See the NOTICE file
+#* distributed with this work for additional information
+#* regarding copyright ownership.  The ASF licenses this file
+#* to you under the Apache License, Version 2.0 (the
+#* "License"); you may not use this file except in compliance
+#* with the License.  You may obtain a copy of the License at
+#*
+#*   http://www.apache.org/licenses/LICENSE-2.0
+#*
+#* Unless required by applicable law or agreed to in writing,
+#* software distributed under the License is distributed on an
+#* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#* KIND, either express or implied.  See the License for the
+#* specific language governing permissions and limitations
+#* under the License.
+#*
+#*************************************************************/
+
+'''
+Example script of CNN model for CIFAR10 dataset
+'''
+import os, sys
+import numpy as np
+
+current_path_ = os.path.dirname(__file__)
+singa_root_=os.path.abspath(os.path.join(current_path_,'../../..'))
+sys.path.append(os.path.join(singa_root_,'tool','python'))
+
+from singa.driver import Driver
+from singa.layer import *
+from singa.model import *
+from singa.utils.utility import swap32
+
+fname_mean_image = 'tool/python/examples/datasets/cifar10_mean_image'
+mean_image = np.fromfile(fname_mean_image)
+
+def load_dataset(did=1):
+    ''' CIFAR10 dataset
+        5 binary dataset, each contains 10000 images
+        1 row (1 image) includes 1 label & 3072 pixels
+        3072 pixels are  3 channels of a 32x32 image
+    '''
+    print '[Load CIFAR10 dataset]', did
+    dataset_dir_ = singa_root_ + "/examples/cifar10/cifar-10-batches-bin"
+    fname_train_data = dataset_dir_ + "/data_batch_{}.bin".format(did)
+    
+    nb_samples = 10000
+    nb_pixels = 3 * 32 * 32  
+    d = np.fromfile(fname_train_data, dtype=np.uint8)
+    d = d.reshape(nb_samples, nb_pixels + 1) # +1 for label
+    x = d[:, 1:] 
+    x = x - mean_image
```

Comment:
It would be better to compute the mean online after loading the data? Currently, we assume that there is a mean file, but it is not known where to get it..
e.g.,
```
    image=...
    label=...
    mean=np.avg(image,...)
    image-=mean
```
There is code for loading cifar10 data (in python format), http://www.cs.toronto.edu/~kriz/cifar.html.
Maybe we can add a usage string telling users where to download the dataset (similar for the mnist dataset).